aws-auth-cognito/src/main/java/com/amplifyframework/auth/cognito/AWSCognitoAuthPlugin.kt

```diff
@@ -828,16 +832,18 @@ class AWSCognitoAuthPlugin : AuthPlugin<AWSCognitoAuthServiceBehavior>() {
             { authState ->
                 when (val authZState = authState.authZState) {
                     is AuthorizationState.SessionEstablished -> {
-                        when (authZState.deleteUserState) {
+                        val deleteUserState = authZState.deleteUserState
+                        when (deleteUserState) {
                             is DeleteUserState.UserDeleted -> {
                                 onSuccess.call()
                                 Amplify.Hub.publish(HubChannel.AUTH, HubEvent.create(AuthChannelEventName.USER_DELETED))
                                 listenerToken?.let(authStateMachine::cancel)
                             }
                             is DeleteUserState.Error -> {
-                                //TODO: Add an appropriate error exception for Delete user to AuthException
-                                onError.accept(AuthException.InvalidAccountTypeException())
                                 listenerToken?.let(authStateMachine::cancel)
+                                onError.accept(
```

Comment:
is there a reason we have onError.accept triggered twice in this logic block? How can we tell whether the account is invalid in line 839?

Reply:
The authstatemachine::cancel will cancel any further calls to the SM.